pkg/limits/service_test.go

```diff
@@ -2,500 +2,204 @@ package limits
-			usage: &usageStore{
-				numPartitions: 2,
-				locks:         make([]stripeLock, 2),
-				stripes: []map[string]tenantUsage{
-					make(map[string]tenantUsage),
-					make(map[string]tenantUsage),
-				},
-			},
-			ActiveWindow:     time.Hour,
-			rateWindow:       5 * time.Minute,
-			BucketSize:       time.Minute,
-			maxActiveStreams: 3,
-			// request data
-			tenantID: "tenant1",
-			streams: []*proto.StreamMetadata{
-				{StreamHash: 0x1, TotalSize: 1010},
-				{StreamHash: 0x2, TotalSize: 1010},
-				{StreamHash: 0x3, TotalSize: 1010},
-				{StreamHash: 0x4, TotalSize: 1010},
-			},
-			// expect data
-			expectedIngestedBytes: 2020,
-			expectedResults: []*proto.ExceedsLimitsResult{
-				{StreamHash: 0x3, Reason: uint32(ReasonExceedsMaxStreams)},
-				{StreamHash: 0x4, Reason: uint32(ReasonExceedsMaxStreams)},
-			},
-			expectedNumRecords: 2,
+		expected: &proto.ExceedsLimitsResponse{
+			Results: []*proto.ExceedsLimitsResult{{
+				StreamHash: 0x2,
+				Reason:     uint32(ReasonExceedsMaxStreams),
+			}},
 		},
-		{
-			name: "skip streams assigned to partitions not owned by instance but enforce limit",
-			// setup data
-			assignedPartitions: []int32{0},
-			numPartitions:      2,
-			usage: &usageStore{
-				numPartitions: 1,
-				locks:         make([]stripeLock, 2),
-				stripes: []map[string]tenantUsage{
-					make(map[string]tenantUsage),
-					make(map[string]tenantUsage),
-				},
-			},
-			ActiveWindow:     time.Hour,
-			rateWindow:       5 * time.Minute,
-			BucketSize:       time.Minute,
-			maxActiveStreams: 3,
-			// request data
-			tenantID: "tenant1",
-			streams: []*proto.StreamMetadata{
-				{StreamHash: 0x1, TotalSize: 1010}, // Unassigned
-				{StreamHash: 0x2, TotalSize: 1010}, // Assigned
-				{StreamHash: 0x3, TotalSize: 1010}, // Unassigned
-				{StreamHash: 0x4, TotalSize: 1010}, // Assigned  but exceeds stream limit
-			},
-			// expect data
-			expectedIngestedBytes: 1010,
-			expectedResults: []*proto.ExceedsLimitsResult{
-				{StreamHash: 0x4, Reason: uint32(ReasonExceedsMaxStreams)},
-			},
-			expectedNumRecords: 1,
+		expectedEntries: 1,
+	}, {
+		// This test asserts that an expired stream is not counted towards
+		// the max active stream limit.
+		name:             "expired stream should be re-refreshed",
+		tenant:           "tenant",
+		activeWindow:     DefaultActiveWindow,
+		rateWindow:       DefaultRateWindow,
+		bucketSize:       DefaultBucketSize,
+		maxActiveStreams: 1,
+		currentUsage: []streamUsage{{
+			hash:        0x1,
+			lastSeenAt:  now.Add(-DefaultActiveWindow - 1).UnixNano(),
+			totalSize:   100,
+			rateBuckets: newRateBuckets(DefaultRateWindow, DefaultBucketSize),
+		}},
+		request: &proto.ExceedsLimitsRequest{
+			Tenant: "tenant",
+			Streams: []*proto.StreamMetadata{{
+				StreamHash: 0x2,
+				TotalSize:  100,
+			}},
 		},
-	}
+		expected: &proto.ExceedsLimitsResponse{
+			Results: make([]*proto.ExceedsLimitsResult, 0),
+		},
+		expectedEntries: 2,
```

Comment:
Shouldn't this be one as per `0x1` expired? Assuming `iterTenant` is counting only active streams now here.

Reply:
It's still counting all streams, both active and expired 👍